### PR TITLE
add current node name in topbar

### PR DESCRIPTION
**Pull request by bendaamerahm**
- putting the name of current node between the title and tabs  "Map"
- truncate node name by adding three dots 
- adding tooltip to the node name that show the entire name text 
- handle all events around the current node (update , delete ..)

### CODE REVIEW

File: src/components/app-top-bar.js [hunk 1]
```diff
@@ -191,8 +207,20 @@ const AppTopBar = ({ user, tabIndex, onChangeTab, userManager }) => {
             fetchAppsAndUrls().then((res) => {
                 setAppsAndUrls(res);
             });
+            if (studyUuid !== null) {
+                fetchNetworkModificationTree(studyUuid)
+                    .then((res) => {
```
Comment (sBouzols) [inline src/components/app-top-bar.js:212]: not sure it's the role of the TopBar Component to fetch NetworkModificationTree...

File: src/redux/reducer.js [hunk 2]
```diff
@@ -48,6 +48,7 @@ import {
     SET_MODIFICATIONS_DRAWER_OPEN,
     FLUX_CONVENTION,
     CENTER_ON_SUBSTATION,
+    CURRENT_NODE,
```
Comment (sBouzols) [inline src/redux/reducer.js:51]: Why not using the already existing WORKING_TREE_NODE ?

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/app-top-bar.js [hunk 3]
```diff
@@ -44,6 +52,12 @@ const useStyles = makeStyles(() => ({
     tabs: {
         marginLeft: 18,
     },
+    sub_title: {
+        marginLeft: 18,
+        alignSelf: 'center',
+        color: `${LIGHT_THEME}`,
```
[the remark this answers is not in the document]
Comment (bendaamerahm) [reply src/components/app-top-bar.js:58]: ill change it  with theme.palette.primary.main

File: src/components/network-modification-tree-pane.js [hunk 4]
```diff
@@ -83,6 +88,43 @@ export const NetworkModificationTreePane = ({
...
     const prevTreeDisplay = usePreviousTreeDisplay(studyMapTreeDisplay, width);
+    const workingNode = useSelector((state) => state.workingTreeNode);
+    //we use this useRef to avoid to trigger on this depedencie workingTreeNode. avoid infinite loop
+    const workingNodeRef = useRef();
+    workingNodeRef.current = workingNode;
+
+    // change working node to ROOT node after deleting the current working node
+    const updateWorkingNodeToRootAfterDeleting = useCallback(() => {
+        const networkModificationTree = fetchNetworkModificationTree(studyUuid);
+
```
Comment (sBouzols) [inline src/components/network-modification-tree-pane.js:99]: Remove this fetch and use `NetworkModificationTreeModel` in the redux which is the TreeModel

File: src/components/network-modification-tree-pane.js [hunk 5]
```diff
@@ -92,6 +134,18 @@ export const NetworkModificationTreePane = ({
                 )
             ).then((values) => {
                 dispatch(networkModificationTreeNodesUpdated(values));
+                // if working node is present in updated values then we save the new values
+                if (values.find(({ id }) => id === workingNodeRef.current.id)) {
+                    const res = values.find(({ id }) => {
```
Comment (sBouzols) [inline src/components/network-modification-tree-pane.js:139]: the find is duplicated

File: src/components/network-modification-tree-pane.js [hunk 6]
```diff
@@ -119,6 +173,16 @@ export const NetworkModificationTreePane = ({
                 studyUpdatedForce.eventData.headers['updateType'] ===
                 'nodeDeleted'
             ) {
+                dispatch(selectTreeNode(null));
+
+                // handle the case of deleting the working node
+                if (
+                    studyUpdatedForce.eventData.headers['nodes'][0] ===
```
Comment (sBouzols) [inline src/components/network-modification-tree-pane.js:180]: // TODO manage multiple delete notification later

File: src/components/network-modification-tree.js [hunk 7]
```diff
@@ -86,6 +81,11 @@ const NetworkModificationTree = ({
             ) {
                 dispatch(workingTreeNode(element));
             }
+            dispatch(
```
Comment (sBouzols) [inline src/components/network-modification-tree.js:84]: restore order

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/components/network-modification-tree.js [hunk 8]
```diff
@@ -144,6 +147,34 @@ const NetworkModificationTree = ({
...
+        const node = treeModel?.treeElements.find(
+            (entry) => entry?.id === workingNodeRef.current?.id
+        );
+
+        if (node) {
+            dispatch(workingTreeNode(node));
+        } else {
+            // handle the case of deleting the working node
+
+            const rootNode = treeModel?.treeElements.find(
```
Comment (sBouzols) [inline src/components/network-modification-tree.js:160]: Use `getFirstNodeOfType` here

**Review by sBouzols — CHANGES_REQUESTED** [review]

**Review by sBouzols — APPROVED** [review]
Code Review ok
Tests done :
* light/dark theme
* tree node navigation
* tabs navigation
* Node renaming synchro
* Node deletion synchro
OK

**Pull request merged**